lib/ipaddr.rb

```diff
@@ -146,12 +194,20 @@ def ~
     return self.clone.set(addr_mask(~@addr))
   end
 
-  # Returns true if two ipaddrs are equal.
+  # Returns true if two ipaddrs are equal. Does not include the
+  # mask_addr in the check.
   def ==(other)
     other = coerce_other(other)
     return @family == other.family && @addr == other.to_i
   end
 
+  # Returns true if two ipaddrs are equal, including their
+  # mask_addrs
+  def equal_with_mask(other)
```

Comment:
I don't like this name.  It should have a `?` suffix, and it might be too long to be handy.  Also, it is not efficient to create two NetMask objects only to compare with each other.  Any thoughts?